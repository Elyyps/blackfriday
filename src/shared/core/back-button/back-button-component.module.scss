.back-button {
  font-family: Assistant;
  font-size: 12px;
  font-weight: 600;
  font-style: normal;
  font-stretch: normal;
  line-height: 1.5;
  letter-spacing: normal;
  color: #999999;
  cursor: pointer;
  align-items: center;
  display: flex;
  flex-direction: row;
  svg{
    transition: transform ease 0.25s;
    margin-right: 8px;
  }
  &:hover {
    svg {
      transform: translateX(-5px);
    }
  }
}
