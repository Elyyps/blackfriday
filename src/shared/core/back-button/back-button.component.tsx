import * as React from "react";
import styles from "./back-button-component.module.scss";
import IconLeft from "@assets/icons/chevron-left.svg";
import { IconComponent } from "../icon";

export interface IBackLinkComponentProps {
  children?: any;
  className?: any;
  href?: string;
}

const BackLinkComponent = (props: IBackLinkComponentProps) => (
  <a href={props.href} className={`${styles["back-button"]} ${props.className}`}>
    <IconComponent icon={IconLeft} size="13px" />
    {props.children}
  </a>
);

export { BackLinkComponent };
