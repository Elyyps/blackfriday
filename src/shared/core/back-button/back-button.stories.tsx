import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { BackLinkComponent } from "./back-button.component";

storiesOf("BackLink", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <BackLinkComponent />);
