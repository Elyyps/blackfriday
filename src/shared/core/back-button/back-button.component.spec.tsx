import React from "react";
import { shallow } from "enzyme";
import { BackLinkComponent } from "./back-button.component";

describe("[BackLink]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<BackLinkComponent />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
