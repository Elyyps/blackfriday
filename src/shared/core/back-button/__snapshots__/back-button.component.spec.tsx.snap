// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[BackLink] should render component without crashing 1`] = `
<a
  className="back-button undefined"
>
  <IconComponent
    icon="chevron-left.svg"
    size="13px"
  />
</a>
`;
